setup.md

```diff
@@ -0,0 +1,92 @@
+### Install mbed-cli to a venv
```

Comment:
Is it really useful to have instructions for installing `mbed-cli` here, rather than linking to https://github.com/ARMmbed/mbed-cli#installing-mbed-cli? See https://github.com/ARMmbed/mbed-os-example-blinky and https://github.com/ARMmbed/mbed-os-example-tls for comparison.

Reply:
Yes and no. We could relegate the venv trick for coping with the tool to oral tradition. If the tool didn't clobber your environment so much, I'd recommend just linking, sure.

Reply:
I've changed it so there's an easy setup mentioned first. If you'd like it to be a link to the instructions  instead Gilles, let me know.